modules/bot-improvement/src/views/full/index.tsx

```
@@ -0,0 +1,199 @@
+import { Callout, Icon } from '@blueprintjs/core'
+import { Tab, Tabs } from '@blueprintjs/core'
+import { Container } from 'botpress/ui'
+import _ from 'lodash'
+import React, { FC, useEffect, useState } from 'react'
+
+import { FeedbackItem, Goal, QnAItem } from '../../backend/typings'
+
+import { makeApi } from './api'
+import Conversation from './components/messages/Conversation'
+import FeedbackItemComponent from './components/FeedbackItem'
+import style from './style.scss'
+
+type SelectedTabId = 'pending' | 'solved'
+
+const defaultState: {
+  feedbackItems: FeedbackItem[]
+  qnaItems: QnAItem[]
+  goals: Goal[]
+  feedbackItemsLoading: boolean
+  currentFeedbackItem: FeedbackItem
+  selectedTab: SelectedTabId
+} = {
+  feedbackItems: [],
+  qnaItems: [],
+  goals: [],
+  feedbackItemsLoading: true,
+  currentFeedbackItem: null,
+  selectedTab: 'pending'
+}
+
+const getDefaultQnaItemId = (qnaItems: QnAItem[]) => {
+  return qnaItems[0].id
+}
+
+const getDefaultGoalId = (goals: Goal[]) => {
+  return goals[0].id
+}
+
+export default props => {
+  const { bp, contentLang } = props
+  const api = makeApi(bp)
+
+  const [state, setState] = useState(defaultState)
+
+  const { qnaItems, goals, feedbackItems, feedbackItemsLoading } = state
+  let { currentFeedbackItem } = state
+
+  useEffect(() => {
+    const initializeState = async () => {
+      const qnaItems = await api.getQnaItems()
+      const goals = await api.getGoals()
+
+      const defaultQnaItemId = getDefaultQnaItemId(qnaItems)
+
+      const feedbackItems = (await api.getFeedbackItems()).map(i => {
+        i.correctedActionType = i.correctedActionType || 'qna'
+        i.correctedObjectId = i.correctedObjectId || defaultQnaItemId
+        i.status = i.status || 'pending'
+        return i
+      })
+
+      setState(state => ({ ...state, feedbackItemsLoading: false, feedbackItems, goals, qnaItems }))
+    }
+    initializeState().catch(e => {
+      throw e
+    })
```

Why the empty catch/throw?